test/log_setup_test.cc

```diff
@@ -0,0 +1,45 @@
+#include <boost/smart_ptr/shared_ptr.hpp>
+#include <boost/smart_ptr/make_shared_object.hpp>
+#include <boost/log/core.hpp>
+#include <boost/log/trivial.hpp>
+#include <boost/log/sinks/sync_frontend.hpp>
+#include <boost/log/sinks/text_ostream_backend.hpp>
+#include <boost/log/sources/logger.hpp>
+#include <boost/log/sources/record_ostream.hpp>
+#include <boost/log/utility/setup/console.hpp>
+
+#include <sstream>
+
+namespace logging = boost::log;
+namespace src = boost::log::sources;
+namespace sinks = boost::log::sinks;
+
+int main(int, char*[])
+{
+    // Construct the sink
+    typedef sinks::synchronous_sink< sinks::text_ostream_backend > text_sink;
+    boost::shared_ptr< text_sink > sink = boost::make_shared< text_sink >();
+
+    auto log_stream = boost::make_shared< std::ostringstream >();
+
+    // Add a stream to write log to
+    sink->locked_backend()->add_stream(log_stream);
+
+    // Register the sink in the logging core
+    logging::core::get()->add_sink(sink);
+
+    boost::log::add_console_log(std::cout, boost::log::keywords::format = "");
```

Comment:
Could also roll this into `log_test.cc` and not make a new test. Maybe that's better?

Reply:
I'd be inclined to roll into one, I think, yeah -- they're quite similar and the differences aren't apparent at a glance.

Let's add a brief comment on the new lines explaining what they're testing, or at least linking to #151, for future reference.